### PR TITLE
11 - Setup Initial File and Folder Structure - Baz

### LINKED ISSUE #11: Setup Initial File & Folder Structure
## Story

- **As a** developer

- **I Want** to have a clearly defined file/folder structure when I am developing the project

- **So That** this helps me know where to put my new code and to help me avoid merge conflicts

## Acceptance Criteria

1. Given I am developing the project
   - When I am developing a specific page or component
   - Then I have a clearly defined file/folder structure to help me know where to write code and to prevent me making merge conflicts

## Implementation Notes

1.

## Checklist:

- [ ] I have defined the user story
- [ ] I have defined some acceptance criteria
- [ ] I have included any implementation notes
- [ ] I have included all relevant links

### PR DESCRIPTION
**Pull request by bazmurphy**
## Description

We agreed as a team during a meeting to create an initial folder/file structure to facilitate easier group work and to help avoid merge conflicts on the Project.

This PR includes:

- adding a set of subfolders to `pages` for each **Page** of the website
  - adding a set of `components` subfolders for each **Page** folder for Page specific components
- adding a global `components` folder - for storing globally reusable components
  -  currently included are Header and Footer
- adding routing to `App.jsx` following react-router-dom 6.4+ practices
- adding `layout` folder - initially using `RootLayout.jsx`
- adding a global `utils` folder - for storing global utility functions
- adding basic nav links to `Header.jsx` to test the aformentioned routing
- adding very basic css reset and Inter google font

## Related to
Issue #11

## How to Test:
1. `git fetch origin` (get the latest changes from the remote repository)
2. `git switch 11-setup-initial-file-folder-structure` (change to the PR branch)
3. `npm run dev:client` (run the client in development mode)
4. visit http://localhost:3000

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [x] ~I have commented my code~ I have written in Markdown files the purpose of the folders that have been added/adjusted
- [x] ~I have updated any documentation~ See above

### CODE REVIEW

File: client/src/App.css [hunk 1]
```diff
@@ -0,0 +1,8 @@
+@import url('https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800;900&display=swap');
+
+* {
```
Comment (ShayanMahnam) [inline client/src/App.css]: I wanted to say do this :D thanks

File: client/src/components/Header/Header.css [hunk 2]
```diff
@@ -0,0 +1,6 @@
+header > nav > ul {
```
Comment (ShayanMahnam) [inline client/src/components/Header/Header.css:1]: we need to talk about this in our team how we want to use selectors in CSS

**Review by ShayanMahnam — APPROVED** [review]
I checked in my local everything is working. Thank you

**Comment by tiritaki** [comment]
great work Baz,
I like it
image

File: client/src/components/Header/Header.css (hunk 2 above)
Comment (bazmurphy) [reply client/src/components/Header/Header.css:1]: yes, i just stuck this in there very quickly (without classnames) so it wasn't a horrible vertical list that everyone had to suffer looking at whilst they were working on each page.

yes, we absolutely need to decide on how we are going to do the css at all:
css, css modules, sass, styled components, (or a library: tailwind, material, chakra etc)

I like the idea of having styles where we do not have to worry about "has that class name already been used?"

Also we need to discuss a "global" stylesheet that applies to everything

Maybe you can make a thread on Slack for the team to think about and discuss this

File: client/src/App.css (hunk 1 above)
Comment (bazmurphy) [reply client/src/App.css]: again i stuck this in so we didn't have to look at an ugly default font.

actually there is a better way to do this in react where we can import the font beforehand so it doesn't need to fetch the font in this way and it comes packaged already in the bundle, 
similar to the nextjs13 feature. 
we can look into that later.

**Comment by bazmurphy** [comment]
I was not entirely sure if i should suffix "Page" on to all of them.. 
But I went for "more specific" so that is absolutely clear.
If you guys just want "About" and not "AboutPage" for example, then let me know.

**Comment by bazmurphy** [comment]
Initially I followed the React Router style that was presented in the starter-kit.
But I have now adjusted it to the newer/better way using React Router 6.4+ practices

**Comment by VitalinaKuzmenko** [comment]
Everything looks great for me! I merge it!

**Pull request merged**